include/onnxruntime/core/session/onnxruntime_ep_c_api.h

```diff
@@ -86,28 +87,64 @@ struct OrtEpApi {
 
   ORT_CLASS_RELEASE(EpDevice);
 
-  /** \brief Specify nodes that are supported by an OrtEp and should be fused into one node.
-   *
-   * IMPORTANT: This is not the final version of this API function. This is currently experimental but will
-   * be stabilized by the ONNX Runtime 1.23 release.
+  /** \brief Create an OrtNodeFusionOptions instance for specifying options for fusing nodes supported by an
+   * execution provider.
    *
    * Because the nodes will be fused into one "fused node", there must not exist an unsupported node in
    * a path between two of the provided nodes. Otherwise, the graph will become invalid.
+   *
+   * \param[in] nodes Array of nodes supported by the EP that should be fused.
+   * \param[in] num_nodes The number of nodes.
+   * \param[out] out Output parameter set to the OrtNodeFusionOptions instance that is created.
+   *
+   * \since Version 1.23.
+   */
+  ORT_API2_STATUS(CreateNodeFusionOptions, _In_reads_(num_nodes) const OrtNode* const* nodes, _In_ size_t num_nodes,
+                  _Outptr_ OrtNodeFusionOptions** out);
```

Comment:
Do we need to put the nodes inside the options? I think the usage would be slightly more natural to keep them separate, which also allows the node fusion options argument to EpGraphSupportInfo_AddNodesToFuse to be optional. 

Do we need an opaque type for this or would a struct with a version number be simpler for this usage? Opaque types are more flexible, but I'm mindful about how large the API surface could get if we make everything an opaque type. In this case the usage seems to be one-way (EP author populates the options, passes it in to ORT, and they're done with it), so I'm not sure we would need to stuff anything extra into it in the background.

Reply:
Updated to make the options struct not opaque. Nodes are also provided to AddNodesToFuse(). It is simpler. Thanks.

If we have to change it, I think we'll find out soon as we port the TRT EP.